CDBTo3DTiles/src/CDBRMDescriptor.cpp

```diff
@@ -0,0 +1,245 @@
+        materialNode = materialNode->next_sibling()) {
+        
+        debugIds.emplace_back(currentId++);
+        rapidxml::xml_node<> *materialNameNode = materialNode->first_node("Name");
+
+        // Iterate through Material(s).
+        rapidxml::xml_node<> *substrateNode = materialNode->first_node("Primary_Substrate");
+        for (rapidxml::xml_node<> *baseMaterialNode = substrateNode->first_node("Material"); baseMaterialNode;
+            baseMaterialNode = baseMaterialNode->next_sibling()) {
+            // Increase substrate offset.
+            substrateOffset++;
+            // Get base material index from name.
+            auto baseMaterial = &materials->baseMaterials.find(baseMaterialNode->first_node("Name")->value())->second;
+            int enumIndex = baseMaterial->value;
+            // Get weight.
+            int weight = std::stoi(baseMaterialNode->first_node("Weight")->value());
+
+            substrates.emplace_back(enumIndex);
+            weights.emplace_back(weight);
+        }
+        // Insert substrate offset.
+        arrayOffsets.emplace_back(substrateOffset);
+
+        // Insert name.
+        std::string name = materialNameNode->value();
+        compositeMaterialNames.emplace_back(std::vector<uint8_t> (name.begin(), name.end()));
+
+        // Update start offset of next name.
+        currentOffset += materialNameNode->value_size();
+        // Insert offset.
+        compositeMaterialNameOffsets.emplace_back(currentOffset);
+    }
+
+    // Get glTF buffer.
+    auto bufferData = &gltf->buffers[0].data;
+    size_t bufferSize = bufferData->size();
+
+    // Setup bufferView for string offsets.
+    size_t offsetsBufferSize = sizeof(uint8_t) * compositeMaterialNameOffsets.size();
+    tinygltf::BufferView offsetsBufferView;
+    offsetsBufferView.buffer = 0;
+    offsetsBufferView.byteOffset = bufferSize;
+    offsetsBufferView.byteLength = offsetsBufferSize;
+    gltf->bufferViews.emplace_back(offsetsBufferView);
+    // Add offsets to buffer.
+    bufferData->resize(bufferSize + offsetsBufferSize);
+    std::memcpy(bufferData->data() + bufferSize, compositeMaterialNameOffsets.data(), offsetsBufferSize);
+    bufferSize += offsetsBufferSize;
+
+    // Add string to buffer.
+    tinygltf::BufferView stringBufferView;
+    stringBufferView.buffer = 0;
+    stringBufferView.byteOffset = bufferSize;
+    stringBufferView.byteLength
+        = currentOffset; // The current offset will point to the end of the string buffer.
+    gltf->bufferViews.emplace_back(stringBufferView);
+    // Add strings to buffer.
+    bufferData->resize(bufferSize + currentOffset);
+    for (auto &stringData : compositeMaterialNames) {
+        std::memcpy(bufferData->data() + bufferSize, stringData.data(), stringData.size());
+        bufferSize += stringData.size();
+    }
+
+    // Setup bufferView for substrates.
+    size_t substratesBufferSize = sizeof(uint8_t) * substrates.size();
```

Comment:
This adding the buffer data and buffer view process is repeated four times. Can we make a function like "addBuffer" that does this for a given buffer/buffer view and call it four times with the four different buffers?

Reply:
Added `createMetadataBufferView` function to `Gltf.cpp`.